tensorflow_lite_support/ios/task/audio/sources/TFLAudioClassifier.h

```diff
@@ -0,0 +1,175 @@
+ limitations under the License.
+ ==============================================================================*/
+#import <Foundation/Foundation.h>
+
+#import "tensorflow_lite_support/ios/task/audio/core/audio_tensor/sources/TFLAudioTensor.h"
+#import "tensorflow_lite_support/ios/task/core/sources/TFLBaseOptions.h"
+#import "tensorflow_lite_support/ios/task/processor/sources/TFLClassificationOptions.h"
+#import "tensorflow_lite_support/ios/task/processor/sources/TFLClassificationResult.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * Options to configure `TFLAudioClassifier`.
+ */
+NS_SWIFT_NAME(AudioClassifierOptions)
+@interface TFLAudioClassifierOptions : NSObject
+
+/**
+ * Base options that are used for creation of any type of task.
+ * @discussion Please see `TFLBaseOptions` for more details.
+ */
+@property(nonatomic, copy) TFLBaseOptions *baseOptions;
+
+/**
+ * Options that configure the display and filtering of results.
+ * @discussion Please see `TFLClassificationOptions` for more details.
+ */
+@property(nonatomic, copy) TFLClassificationOptions *classificationOptions;
+
+/**
+ * Initializes a new `TFLAudioClassifierOptions` with the absolute path to the model file stored
+ * locally on the device.
+ *
+ * @discussion The external model file, must be a single standalone TFLite file. It could be packed
+ * with TFLite Model Metadata[1] and associated files if exist. Fail to provide the necessary
+ * metadata and associated files might result in errors. Check the
+ * [documentation](https://www.tensorflow.org/lite/convert/metadata) for each task about the
+ * specific requirement.
+ *
+ * @param modelPath An absolute path to a TensorFlow Lite model file stored locally on the device.
+ *
+ * @return An instance of `TFLAudioClassifierOptions` initialized to the given model path.
+ */
+- (instancetype)initWithModelPath:(NSString *)modelPath;
+
+- (instancetype)init NS_UNAVAILABLE;
+
++ (instancetype)new NS_UNAVAILABLE;
+
+@end
+
+/**
+ * A TensorFlow Lite Task Audio Classifiier.
+ */
+NS_SWIFT_NAME(AudioClassifier)
+@interface TFLAudioClassifier : NSObject
+
+/**
+ * Initializes a new `TFLAudioClassifier` with the absolute path to the model file stored locally on
+ * the device.
+ *
+ * @discussion The external model file, must be a single standalone TFLite file. It could be packed
+ * with TFLite Model Metadata[1] and associated files if exist. Fail to provide the necessary
+ * metadata and associated files might result in errors. Check the
+ * [documentation](https://www.tensorflow.org/lite/convert/metadata) for each task about the
+ * specific requirement.
+ *
+ * @param modelPath Path to the model file.
+ *
+ * @param error An optional error parameter populated if there is an error in initializing the audio
+ * classifier.
+ *
+ * @return A `TFLAudioClassifier` instance if initialization is successful or nil in case of
+ * failure.
+ */
+- (nullable instancetype)initWithModelPath:(NSString *)modelPath error:(NSError **)error;
```

Comment:
Remove this to align with the ImageClassifier API.
https://github.com/tensorflow/tflite-support/blob/master/tensorflow_lite_support/ios/task/vision/sources/TFLImageClassifier.h